synedrion/src/uint/signed.rs

```diff
@@ -411,8 +381,8 @@ where
 {
     /// Returns a [`Signed`] with the same value, but twice the bit-width.
     /// Consumes `self`, but under the hood this method clones.
-    pub fn into_wide(self) -> Signed<T::Wide> {
-        let abs_result = self.abs().into_wide();
+    pub fn to_wide(self) -> Signed<T::Wide> {
```

Comment:
Strange that clippy didn't complain about it, because it did in other seemingly identical situations with other types

Reply:
Fixed

Reply:
Actually no. The reason clippy doesn't complain is because `Signed` here implements `Copy`, so `to_*(self)` is appropriate. It implements `Copy` because of the `ConditionallySelectable` bound. So the name is fine as it is.

The problem is that it uses `abs()` inside which creates a copy on the stack that we don't really want, and `abs()` is used in a lot of methods which really shouldn't copy the number. I wonder if `Signed` can be rewritten to remove all this copying (and ideally the whole `ConditionallySelectable` bound).

Reply:
Filed https://github.com/entropyxyz/synedrion/issues/165 to not inflate this PR any more